typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -393,6 +403,20 @@ export abstract class HyperlaneDeployer<
       constructorArgs,
       deployOpts,
     );
+    // If the proxy already existed in artifacts but the implementation did not,
+    // we only deploy the implementation and keep the proxy.
+    // Changing the proxy's implementation address on-chain is left to
+    // the govern / checker script
```

Comment:
consider adding a separate flag for upgrades instead of implicit "no cached implementation/proxy"